python/ccxt/southxchange.py

```diff
@@ -252,6 +253,18 @@ def fetch_open_orders(self, symbol=None, since=None, limit=None, params={}):
         response = self.privatePostListOrders(params)
         return self.parse_orders(response, market, since, limit)
 
+    def fetch_order(self, id, symbol=None, since=None, limit=None, params={}):
+        self.load_markets()
+        market = None
+        if symbol is not None:
+            market = self.market(symbol)
+        response = self.privatePostListOrders(params)
+        orders = self.parse_orders(response, market, since, limit)
+        for order in orders:
+            if id.find(order['id']) != -1:
```

Comment:
id == order['id']

Reply:
Order ID is written to db with extra characters for some reason. Instead of trying to strip extra characters, we simply look for the order substring.